StudentPlan.py

```diff
@@ -5,9 +5,18 @@ class StudentPlan:
 
     semesters = []
 
-    def __init__(self, curriculum, max_credits):
+    def __init__(self, curriculum, max_credits, past_semesters=None):
         self.curriculum = curriculum
         self.max_credits = max_credits
+        if past_semesters is not None:
+            for semester_position in range(max(past_semesters.keys())+1):
+                past_semester = self.add_semester(semester_position, past=True)
+                try:
+                    courses_to_add = past_semesters[semester_position]
```

Comment:
You can use the dictionary method get() and specify a default value, as a second parameter, if semester_position doesn't exist in past_semester. So you wouldn't have to use the try/except block.

Also, you can just check if the the key semester_position is in the dictionary past_semester.

Reply:
Done